packages/flutter/lib/src/rendering/box.dart

```diff
@@ -471,6 +471,28 @@ class BoxParentData extends ParentData {
 /// ContainerRenderObjectMixin.
 abstract class ContainerBoxParentDataMixin<ChildType extends RenderObject> extends BoxParentData with ContainerParentDataMixin<ChildType> { }
 
+enum _IntrinsicDimension { minWidth, maxWidth, minHeight, maxHeight }
+class _IntrinsicDimensionsCacheEntry {
+  _IntrinsicDimensionsCacheEntry(this.dimension, this.argument);
+
+  final _IntrinsicDimension dimension;
+  final double argument;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
```

Comment:
We should probably do an audit. I agree that this is a case where we shouldn't have it. It makes sense to have it when you're frequently reusing objects, or when the objects are really complex (ThemeData, for instance).

Filed https://github.com/flutter/flutter/issues/4447